cmd/common/snapshot/process.go

```diff
@@ -35,6 +38,85 @@ type ProcessParser struct {
 	outputConfig *commonutils.OutputConfig
 }
 
+type nodeEvent struct {
+	event    *types.Event
+	children []*nodeEvent
+	parent   *nodeEvent
+}
+
+func createTree(processes []*types.Event) *nodeEvent {
+	nodes := make(map[int]*nodeEvent, 0)
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeEvent{
+			event:    process,
+			children: make([]*nodeEvent, 0),
+			parent:   nil,
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		for _, n := range nodes {
+			if node.event.Pid == n.event.ParentPid {
+				node.children = append(node.children, n)
+				n.parent = node
+			}
+		}
+	}
+
+	// Find the root, i.e. find the node without parent.
+	for _, node := range nodes {
+		if node.parent == nil {
+			return node
+		}
+	}
+
+	return nil
```

Comment:
I think all this can be simplified:

```suggestion
	// Link all nodes together.
	var root *nodeEvent
	for _, node := range nodes {
		ppid := node.event.ParentPid
		if _, ok := nodes[ppid]; !ok {
			root = node
			continue
		}
		nodes[ppid].children = append(nodes[ppid].children, node)
	}

	if root == nil {
		// TODO: return an error instead of panicking.
		panic("container without root process")
	}

	return root
```

BTW, we wouldn't need the `parent` field in `nodeEvent`.

Did I miss something?